test/test_events.py

```diff
@@ -332,8 +333,14 @@ def test_service_pooled_events(rabbit_manager, rabbit_config,
     assert len(bar_queue['consumer_details']) == 1
 
     exchange_name = "srcservice.events"
-    rabbit_manager.publish(vhost, exchange_name, 'eventtype', '"msg"',
-                           properties=dict(content_type='application/json'))
+    rabbit_manager.publish(
+        vhost, exchange_name, 'eventtype', '"msg"',
+        properties=dict(content_type='application/json')
+    )
+    # can't use the entrypoint waiter here because we don't know
+    # which of the "foo" containers will pick up the message,
+    # so just wait for events to propagate
+    time.sleep(.1)
```

Comment:
It's a bad test, because we need to verify that exactly two events are received -- not one or three. The only way to do that is give them a fair chance to propagate through the system. For absolute certainty I should also be checking that all the queues are empty.

For this little irritation I'm not sure it's worth all the extra effort to refactor.